ushahidi/platform-client
# 3781 validating img

This pull request makes the following changes:
- Moves the deletion of an image to the save-step instead of instant deletion. It prevents the scenario "if a user deletes an image, and then change their mind and leave without saving, their image is still deleted"

Testing checklist:
- Select a post with an image
- Delete the image
- Do not save the post, navigate away or click cancel
- [ ] Go to the post again, the image should still be there
- Select the post again and edit it
- Delete the image
- Save the post
- [ ] Go to the post again, the image is deleted!
- [ ] Repeat above with a survey with several images


Fixes ushahidi/platform#3781 (part of it) .

Ping @ushahidi/platform
